benchmarks/partials/partial_dsl.rb

```diff
@@ -0,0 +1,41 @@
+require 'benchmark/ips'
+require 'benchmark/memory'
+require_relative 'setup'
+
+POST_PARTIAL = <<-JBUILDER
+  json.extract! post, :id, :body
+JBUILDER
+
+PARTIALS = { "_post.json.jbuilder" => POST_PARTIAL }
+
+Post = Struct.new(:id, :body)
+
+view = build_view(fixtures: PARTIALS)
+json = JbuilderTemplate.new view
```

Comment:
This approach is slightly leaner than what is happening under test. `JbuilderTemplate` just needs a context to do partial lookups. 

Test are doing a full `render` lifecycle which requires compilation of the `json.partial! "post", post: post` template, and also `target!`s the builder which runs `to_json`, and _then_ serializes it all to a string.

The benchmarks saves on doing this full render cycle and just measures interaction with the builder itself.